text/0249-v2-experiments.md

```diff
@@ -408,3 +484,240 @@ Disadvantages:
+dependency in `setup.py`, but then we also lose the ability for libraries to
+specify their dependency version, which is not really acceptable.
+
+Effectively, this seems to be a no-go.
+
+**.NET Core/NuGet**
+
+No need to change the sources when using a stable library with an
+experimental app, as the namespaces and class names will be the same.
+
+...pending responses from .NET SDK team...
+
+**Go**
+
+???
+
+## Prerelease tags
+
+We would vend the experimental build under a prerelease tag, for example:
+
+- `aws-cdk-lib@1.60.0`
+- `aws-cdk-lib@1.60.0-experimental`
+
+We could get rid of some of the ordering and warning problems by making sure the
+experimental version semver-orders *after* the stable version:
+
+- `aws-cdk-lib@1.60.100-experimental`
+- `aws-cdk-lib@1.61.0-experimental`
+
+But that is most likely just going to break people's heads so let's not even try.
+
+| .     | Stable apps | Experimental app uses Stable lib | Without complaints | Lib advertises Experimental |
+|-------|-------------|----------------------------------|--------------------|-----------------------------|
+| NPM   | yes         | yes                              | no                 | no                          |
+| Maven | no          | with excluded dependencies       | yes                | yes                         |
+| pip   | yes         | yes                              | no                 | yes                         |
+| NuGet | ?           | ?                                | ?                  | ?                           |
+| Go    | ?           | ?                                | ?                  | ?                           |
+
+**NPM**
+
+Stable ranges: yes, `^1.60.0` does not include `1.61.1-experimental`.
+
+Experimental app uses stable lib: can be done, the lib only uses a peerDependency. `npm ls` will complain about
+a missing peerDependency though.
+
+**Maven**
+
+Stable ranges: no, `[1.60.0,)` *will* include `1.61.1-experimental`.
+
+Experimental app uses stable lib:
+
+**pip**
+
+Stable ranges: yes, `>=1.60.0` will not match version `1.60.1-experimental` [ref](https://pip.pypa.io/en/stable/reference/pip_install/#pre-release-versions)
+
+Experimental app uses stable lib: `pip` allows overriding transitive dependencies using `requirements.txt`, but it will complain
+while doing so:
+
+Example (I tried with different packages):
+
+```text
+awscli 1.18.158 has requirement s3transfer<0.4.0,>=0.3.0, but you'll have s3transfer 0.2.0 which is incompatible.
+```
+
+**NuGet**
+
+...pending responses from .NET SDK team...
+
+## Different major versions
+
+We would vend the experimental build under a prerelease tag, for example:
+
+- `aws-cdk-lib@2.60.0`
+- `aws-cdk-lib@3.60.0` (odd ranges are experimental)
+
+Or:
+
+- `aws-cdk-lib@2.60.0`
+- `aws-cdk-lib@102.60.0` (100+ versions are experimental)
+
+| .     | Stable apps | Experimental app uses Stable lib | Without complaints | Lib advertises Experimental |
```

Comment:
I gotta say, the "Without complaints" category is hard, because you have to remember it's only for the  "experimental app with stable lib" scenario. Maybe call it "...without complaints", to make it more obvious?